bpf/libbpf/libbpf_api.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "libbpf.h"
+#include <linux/limits.h>
+#include <net/if.h>
+#include <bpf.h>
+#include <stdlib.h>
+#include <errno.h>
+
+static void set_errno(int ret) {
+	errno = ret >=0 ? 0 : -ret;
```

Comment:
```suggestion
	errno = ret >= 0 ? ret : -ret;
```
as non-zero is always an error and the polarity depends on libbpf version?